### ISSUE TITLE
Print Test Page prints test label followed by blank pages and media becomes misaligned

### ISSUE THREAD

**Issue by tarikjn** [michaelrsweet/lprint#95]
Printer: Zebra ZD420 203dpi DT
Driver: `zpl_4inch-203dpi-dt`
Paper: 4 x 6

Print Test Page feature from the web server prints a test label followed by 4 blank pages at normal speed and ~2.25 blank pages at slow speed. After this, the media alignment must be reset.

This also happens intermittently with other documents but seems to be consistently reproducible using Print Test Page. 

This issue is not present using macOS 13.1 built-in CUPS driver.

**Comment by michaelrsweet** [comment]
What version of LPrint are you using?

**Comment by michaelrsweet** [comment]
Closing for lack of response; will re-open as needed.

**Comment by michaelrsweet** [comment]
Can you attach your config file (~/.config/lprint.conf, ~/Library/Application Support/lprint.conf, /Library/Application Support/lprint.conf)?

Also, have you done any label configuration/calibration on the printer itself?

**Comment by michaelrsweet** [comment]
Quick question - are you using labels with a black mark between labels on the carrier tape? The picture looks like they are just separated by a gap, but the configuration is for a mark. Since LPrint is telling the printer to look for a mark, that is probably why it is feeding extra labels.

**Comment by prohtex** [comment]
Hi @michaelrsweet, I'm very excited to learn about LPrint and have begun using it for our many GK420D printers. First, please let me say "thank you!" for all your hard work on this project.

I am having the issue @tarikjn mentions here. It is worth noting that we're moving from the Allensby/Peninsula drivers. These drivers have a software "calibrate paper size" feature that seems to be independent from the printer's own paper calibration.

Here's what happens:

1. Install 1.3.1 on macOS Sequoia 15.1
2. Visit localhost:8000 and install the printer
3. Media is set to 4x6 label, Gap, Labels
4. Print test page-behavior is as described above; text page prints fine, after which 2-3 labels advance, and the printer fails to sense gap. Eventually stops in the middle of a label with the red light on
5. Reset the printer label sensing (hold down the advance button until two flashes, release, and printer recalibrates the label size)

From what I understand, the printer label size calibration is not enough. There must be some software calibration as well for label size. We've reproduced this on multiple computers and printers, and I've tried all the settings I can think of in terms of Printing Defaults, etc. I thought at first that the page size might be the culprit (ie, printer was being sent something slightly larger than 4x6) but I'm really stumped.

Happy to include any configuration files as needed. I should mention that I do not see any of these files on the system:
~/.config/lprint.conf
~/Library/Application Support/lprint.conf
/Library/Application Support/lprint.conf

**Comment by prohtex** [comment]
![IMG_0546](https://github.com/user-attachments/assets/0d51a6af-4833-42fe-9365-a2f640f78f5b)

**Comment by michaelrsweet** [comment]
@prohtex Just to confirm, you have configured LPrint with 4x6 labels, and the photo you attached are the labels you are using?

**Comment by prohtex** [comment]
> @prohtex Just to confirm, you have configured LPrint with 4x6 labels, and the photo you attached are the labels you are using?

Hi Michael, thanks for the reply. I should have noted in this instance I tried a new label size to troubleshoot. I changed the settings for the smaller labels. At this point I’ve tried every conceivable setting-different sizes, printer models, computers, etc, to no avail!

**Comment by brianbarr** [comment]
I have experienced the same label feed issues. I have a large assortment of vintage zebra printers -- lprint 1.3.1 to a LP2844 using EPL2 mode works fine. lprint to a LP2844-Z does the label feed error thing as described using ZPL mode. I have not tested the LP2844-Z in EPL2 mode. That might be a useful data point.  If I'm not mistaken most ZPL printers will also support EPL2.  Something to try.

**Comment by prohtex** [comment]
Hi @brianbarr, did you have any luck with this? I haven't been able to get anything at all to work, including EPL.

**Comment by msnw84** [comment]
I had this issue for a while with the GK420d, the default option "media-tracking=continuous" was the cause of the problem for me. I changed it to "media-tracking=web" and it's been perfect since.

Hope this helps.

**Comment by zdohnal** [comment]
Yep, I solved this in the past the same way as @msnw84 - media-tracking=web, media-type=labels.

**Comment by wt** [comment]
It looks to me like most people are reporting that changing the media-tracking to something else fixes the problem. FWIW, I have not seen this on two other printers that speak ZPL. I am wondering if it might be appropriate to close this bug for now.

@michaelrsweet Also, is there a feature to capture the data sent to the printer into a text file? I don't mean to configure a file as a printer. I want to duplicate the data sent to the printer to a text file so that I can then manually send it to the printer without using lprint for debugging purposes.

**Comment by prohtex** [comment]
> It looks to me like most people are reporting that changing the media-tracking to something else fixes the problem. FWIW, I have not seen this on two other printers that speak ZPL. I am wondering if it might be appropriate to close this bug for now.
> 
> [@michaelrsweet](https://github.com/michaelrsweet) Also, is there a feature to capture the data sent to the printer into a text file? I don't mean to configure a file as a printer. I want to duplicate the data sent to the printer to a text file so that I can then manually send it to the printer without using lprint for debugging purposes.

I have been desperately trying to solve this for a long time. I'm a software developer and I've tried everything. Changing media tracking does not solve the issue for me.

**Comment by prohtex** [comment]
> [@prohtex](https://github.com/prohtex) Are you in the OpenPrinting telegram chat, by chance?

I am now

**Comment by michaelrsweet** [comment]
Hmm, I've tried to reproduce this on my ZD621d (300dpi), GX420d (203dpi), and GX430t (203dpi), and none of them seem to have this issue no matter what size labels I load.

Going to close this for now in the interests of getting 1.4 out, will re-open if new data is available then...
